### ISSUE TITLE
Odin: Inserting Emoji does not work (most of the times)

### ISSUE THREAD

**Issue by marbetschar** [cassidyjames/ideogram#56]
Seems like something is broken on Odin. Emoji's are not inserted most of the time - regardless which app is used. For whatever reason, the only place which seems to allow pasting Emoji's reliably is the search field in Files. In other places it does not work. Even if I manually execute `Ctrl + v` after selecting an Emoji, it does not get pasted (seems like its never transferred to clipboard, since sometimes earlier copied text gets pasted).

FWIW: I compiled from master and tested in elementary Code, Files, Tasks as well as in VSCodium, Firefox, Signal, Slack, ...

**Comment by leggettc18** [comment]
I've taken a stab at fixing it, read the pull request for full details. What seems to be happening is that the app is closing before the `paste ()` function finishes, clearing the clipboard in the process. Ideally we could save it to a global clipboard or something, but for some reason I was unable to get any of the various methods of storing the clipboard data so it can be accessed after the app closes to work.

The pull request I submitted adds a timeout in front of the `close ()` calls to give the `paste ()` function time to go through. I wrote my own version of the app that essentially reimplements the same functionality in case @cassidyjames is unable to merge the pull request. I originally built it to figure out where the issue came into play, and got most of the way through before I figured out what was happening.
